ydb/_topic_reader/topic_reader_sync.py

```diff
@@ -86,6 +86,57 @@ def async_wait_message(self) -> concurrent.futures.Future:
 
         return self._caller.unsafe_call_with_future(self._async_reader.wait_message())
 
+    def _make_batch_slice(
```

Comment:
**IMPORTANT**

After applying the function, a caller will lose messages that have been trimmed from the batch and will not see these messages in the read session. A server does not allow to skip messages during commit. This can cause problems:
1. If the caller commits messages with ack, the software will hang up forever (because the server will wait for skipped messages before ack the commit).
2. If the caller commits messages without ack. After reconnecting all messages after the last successfully commit (first batch with cut messages) will be re-read. A log of extra work is required to re-read these messages and real progress will be very slow.
3. If the progress is saved on the user's side and messages are not committed to the SDK, the will be lost and cannot be recovered.

Reply:
Ok, I see, though I'm not quite sure I fully understand the path to a solution.
What was the expected approach to take?